common/src/main/java/com/wynntils/commands/QuestCommand.java

```diff
@@ -0,0 +1,271 @@
+import net.minecraft.ChatFormatting;
+import net.minecraft.commands.CommandSourceStack;
+import net.minecraft.commands.Commands;
+import net.minecraft.commands.SharedSuggestionProvider;
+import net.minecraft.network.chat.ClickEvent;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.HoverEvent;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.world.phys.Vec3;
+
+public class QuestCommand extends CommandBase {
+    private static final SuggestionProvider<CommandSourceStack> QUEST_SUGGESTION_PROVIDER =
+            (context, builder) -> SharedSuggestionProvider.suggest(
+                    Managers.Quest.getQuests(QuestSortOrder.ALPHABETIC).stream().map(QuestInfo::getName), builder);
+
+    private static final SuggestionProvider<CommandSourceStack> SORT_SUGGESTION_PROVIDER =
+            (context, builder) -> SharedSuggestionProvider.suggest(
+                    Arrays.stream(QuestSortOrder.values())
+                            .map(order -> order.name().toLowerCase(Locale.ROOT)),
+                    builder);
+
+    @Override
+    public LiteralArgumentBuilder<CommandSourceStack> getBaseCommandBuilder() {
+        return Commands.literal("quest")
+                .then(Commands.literal("list")
+                        .executes((ctxt) -> listQuests(ctxt, "distance"))
+                        .then(Commands.argument("sort", StringArgumentType.word())
+                                .suggests(SORT_SUGGESTION_PROVIDER)
+                                .executes((ctxt) -> listQuests(ctxt, ctxt.getArgument("sort", String.class)))))
+                .then(Commands.literal("search")
+                        .then(Commands.argument("text", StringArgumentType.greedyString())
+                                .executes(this::searchQuests)))
+                .then(Commands.literal("info")
+                        .then(Commands.argument("quest", StringArgumentType.greedyString())
+                                .suggests(QUEST_SUGGESTION_PROVIDER)
+                                .executes(this::questInfo)))
+                .then(Commands.literal("track")
+                        .then(Commands.argument("quest", StringArgumentType.greedyString())
+                                .suggests(QUEST_SUGGESTION_PROVIDER)
+                                .executes(this::trackQuest)))
+                .then(Commands.literal("untrack").executes(this::untrackQuest))
+                .then(Commands.literal("wiki")
+                        .then(Commands.argument("quest", StringArgumentType.greedyString())
+                                .suggests(QUEST_SUGGESTION_PROVIDER)
+                                .executes(this::lookupOnWiki)))
+                .executes(this::syntaxError);
+    }
+
+    private int listQuests(CommandContext<CommandSourceStack> context, String sort) {
+        QuestSortOrder order = QuestSortOrder.fromString(sort);
+
+        Managers.Quest.rescanQuestBook(true, false);
```

Comment:
This method is technically async, and the command doesn't "await" it?

Reply:
Yes. I made some botched efforts to synchronize on this, and then gave up, and just warned the user to retry the command. 

The "proper" solution will require quite some amount of plumbing, and I could not really motivate it. If you think this is too ugly, I'd rather drop the entire /quest command. I'm not sure how useful it will be in practice. I originally started this PR to have a way to exercise the quest model, and I did not want to make a GUI.

Reply:
Okay, this is fine.

Reply:
Thanks.